File: backend/Punishments/Models/ModCaseTemplateExpanded.cs
using Bot.Models;
using Discord;

namespace Punishments.Models;

public class ModCaseTemplateExpanded(ModCaseTemplate template, IUser creator, IGuild guild)
{
    public ModCaseTemplate CaseTemplate { get; set; } = template;
    public DiscordUser Creator { get; set; } = DiscordUser.GetDiscordUser(creator);
    public DiscordGuild Guild { get; set; } = DiscordGuild.GetDiscordGuild(guild);
}

File: frontend/dexter/src/app/models/ModCaseTemplateExpanded.ts
import { DiscordGuild } from "./DiscordGuild";
import { DiscordUser } from "./DiscordUser";
import { ModCaseTemplate } from "./ModCaseTemplate";

export interface ModCaseTemplateExpanded {
    caseTemplate: ModCaseTemplate;
    creator?: DiscordUser;
    guild: DiscordGuild;
}
